src/dataframe/io.jl

```diff
@@ -505,9 +506,33 @@ function builddf(rows::Integer,
                  fields::Integer,
                  p::ParsedCSV,
                  o::ParseOptions)
-    columns = Array(Any, cols)
+    # Determine column names
+    if isempty(o.names)
+        names = gennames(cols)
+    else
+        names = copy(o.names)
+    end
+    # Determine columns to be read from usecols-option
+    # If option not set: read all available columns
+    if isempty(o.usecols)
+        colindices = 1:cols
+    else
+        if isa(o.usecols, Vector{Int})
+            colindices = indexin(o.usecols, 1:cols)
+        else
+            colindices = indexin(o.usecols, names)
+        end
+        if any(colindices .== 0)
+            error("Invalid column selection")
```

Comment:
Could you make the error message more detailed? I know it's terribly frustrating to get this general error instead of the name or index of the incorrect column identifier. Mentioning `usecols` would also be helpful.